services/BasicTypes.proto

```diff
@@ -340,24 +340,50 @@ message CurrentAndNextFeeSchedule {
     FeeSchedule nextFeeSchedule = 2; // Contains next Fee Schedule
 }
 
+/* This contains the IP Address and the port representing an endpoint of a Node in a network */
+message NodeEndpoint {
+    string ipAddress = 1; // The IP address of the node
+    string port = 2; // The port of the node
+}
+
 /*
 The metadata for a Node – including IP Address, and the crypto account associated with the Node.
 */
 message NodeAddress {
-    bytes ipAddress = 1; // The ip address of the Node with separator & octets
-    int32 portno = 2; // The port number of the grpc server for the node
-    bytes memo = 3; // The memo field of the node (usage to store account ID is deprecated)
+    bytes ipAddress = 1 [deprecated=true]; // The ip address of the Node with separator & octets
+    int32 portno = 2 [deprecated=true]; // The port number of the grpc server for the node
+    bytes memo = 3 [deprecated=true]; // The memo field of the node (usage to store account ID is deprecated)
     string RSA_PubKey = 4; // The RSA public key of the node
     int64 nodeId = 5; // A non-sequential identifier for the node
     AccountID nodeAccountId = 6; // The account to be paid for queries and transactions sent to this node
     bytes nodeCertHash = 7; // A hash of the X509 cert used for gRPC traffic to this node
+    repeated NodeEndpoint nodeEndpoint = 8; // A node's IP address and port
+    string description = 9; // A description of the node. Max 100 bytes.
+    int64 stake = 10; // The amount of tinybars staked to this node
 }
 
 /*
 A list of nodes and their metadata.
 */
 message NodeAddressBook {
-    repeated NodeAddress nodeAddress = 1; // Contains multiple Node Address for the network
+    repeated NodeAddress nodeAddress = 1; // Contains all details of the nodes for the network
+}
+
+/* A version of NodeAddress to allow backward compatibility.*/
+message NodeAddressAbbreviated {
+    bytes ipAddress = 1; // The IP address of the Node with separator & octets
+    int32 portno = 2; // The port number of the grpc server for the node
+    bytes memo = 3; // The memo field of the node (usage to store account ID is deprecated)
+    string RSA_PubKey = 4; // The RSA public key of the node
+    int64 nodeId = 5; // A non-sequential identifier for the node. This value is the key between the full and abbreviated address books
+    AccountID nodeAccountId = 6; // The account to be paid for queries and transactions sent to this node
+    bytes nodeCertHash = 7; // A hash of the X509 cert used for gRPC traffic to this node
```

Comment:
In Nosh's writeup in the original ticket, he indicated `nodeCertHash` is deprecated in the abbreviated message. I don't think that is the case since it is currently in the 0.0.101 file and is useful for clients to connect. Can you confirm if it should be present here or not?

Reply:
Removed `deprecated=true` after yesterday's discussion.